pkg/api/types.go

```diff
@@ -1007,6 +1007,12 @@ type PodSecurityContext struct {
 	// Use the host's ipc namespace.
 	// Optional: Default to false.
 	HostIPC bool `json:"hostIPC,omitempty"`
+
+	// SupplementalGroups can be used to specify a list of
+	// additional groups which the main container process will run
+	// as. This will be applied to all containers in the pod in
+	// addition to the primary group of the cotainer.
+	SupplementalGroups []string `json:"supplementalGroups,omitempty"`
```

Comment:
So far we have gone with the pattern of make UIDs ints (see RunAsUser).  Do we want to continue the pattern here or allow for string groups?  

@pmorie @smarterclayton